waku/v2/peermanager/peer_connector.go

```diff
@@ -236,24 +240,35 @@ func (c *PeerConnectionStrategy) dialPeers() {
 
 	for {
 		select {
-		case pd, ok := <-c.GetListeningChan():
-			if !ok {
-				return
-			}
-			addrInfo := pd.AddrInfo
-
-			if addrInfo.ID == c.host.ID() || addrInfo.ID == "" ||
-				c.host.Network().Connectedness(addrInfo.ID) == network.Connected {
+		case <-c.Context().Done():
+			return
+		default:
+			if c.isPaused() {
+				time.Sleep(100 * time.Millisecond) // Sleep for a while to avoid busy waiting
```

Comment:
I am wondering if we should take an approach of read from channel and discard in case connector is paused. 

Anyways once connectivity is resumed, peers would be discovered and connectivity to them would be triggered again from either peerManager or status-go.

Reply:
Sounds good. That way i can avoid having this `default:` case